gr-blocks/lib/file_source_impl.cc

```diff
@@ -260,13 +260,19 @@ int file_source_impl::work(int noutput_items,
 
         uint64_t nitems_to_read = std::min(size, d_items_remaining);
 
-        // Since the bounds of the file are known, unexpected nitems is an error
-        if (nitems_to_read != fread(o, d_itemsize, nitems_to_read, (FILE*)d_fp))
+        size_t nitems_read = fread(o, d_itemsize, nitems_to_read, (FILE*)d_fp);
+        if (nitems_to_read != nitems_read) {
+            // EOF reached for non seekable file, return
+            if (!d_seekable && feof((FILE*)d_fp))
```

Comment:
while this is nice and compact, we typically use { } around conditional blocks; small issue.